packages/nimble-components/src/table/specs/table-spec.md

```diff
@@ -0,0 +1,242 @@
+# Nimble Table
+
+## Overview
+
+The `nimble-table` is a component that offers a way to render tabular data in a variety of ways in each column.
+
+### Background
+
+[Nimble issue #283](https://github.com/ni/nimble/issues/283)
+
+[Nimble Table Research Document](https://lucid.app/lucidspark/31f2314d-dd8e-46fd-8fc1-6e9f66700bb3/edit?viewport_loc=-1060%2C-25492%2C20822%2C12325%2CloaYwcZLRray4&invitationId=inv_38839ad5-72b2-4975-ab7a-6d8be33c960c)
+
+[Table visual design](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/b9cee5e2-49a4-425a-9ed4-38b23ba2e313/)
+
+### Non-goals
+
+-   The `nimble-table` will not offer spreadsheet-like features such as infinite columns, or summary functions.
+
+-   We do not intend for the `nimble-table` to handle data sets of 50K or more on the client-side. Visualizing such sizes of data sets are meant to be enabled through various "data virtualiazation" means, such as the server only sending over subsets of the data at a time, appropriately filtered down to a reasonable size.
+
+-   Non-virtualized static tables.
+
+### Features
+
+In addition to simply rendering data, the `nimble-table` will provide APIs/interactions that will allow a user to do the following:
+
+-   Interactively and programmatically size columns
+-   Interactively and programmatically sort columns
+-   Pin columns
+-   Provide column ordering (programmatically only at first possibly)
+-   Group rows
+-   Multi-select rows (interactively only at first possibly)
+-   Provide hierarchy via the data
+-   Provide custom column rendering
+-   Provide API for adding action menu
+
+#### _Performance_
+
+We intend for the `nimble-table` to handle both the rendering and interactive operations, such as sorting, in a near-instantaneous fashion on datasets of at least 10K rows. Between 10K and 100K rows of data on the client-side, however, we expect to see a notable drop in performance.
+
+Clients, at least initially, will be expected to create mechanisms to restrict the amount of data coming to the client-side.
+
+#### _Out of Scope of Initial Release_
+
+The following are features that we intend to prioritize eventually after the initial release:
+
+-   Hierarchy and Grouping together
```

Comment:
It's probably not worth figuring out now, but my guess is that it'll be just as easy (if not easier) to support this than to figure out what to do with an "invalid" configuration that specifies grouping & hierarchy.

Reply:
Made a slight change to suggest that we could still do this if it were simple enough.